### PR TITLE
Incongruent direction indentation

### PR DESCRIPTION
**Pull request by lucieleblanc**
(Also added a missing check in the congruent-directions case.)

Note that bidirectional text currently doesn't wrap correctly -- see pop-os/cosmic-text#252 for details. This PR only ensures that the first line accounts for indentation.

### CODE REVIEW

File: src/shape.rs [hunk 1]
```diff
@@ -1080,13 +1080,19 @@ impl ShapeLine {
...
+                            current_visual_line.w + word_width > line_width)
                         {
                             // Commit the current line so that the word starts on the next line.
                             if word_range_width > 0.
-                                && wrap == Wrap::WordOrGlyph
-                                && word_width > line_width
+                                && ((wrap == Wrap::WordOrGlyph && word_width > line_width)
+                                    || (wrap == Wrap::WordOrGlyph
+                                        && visual_lines.len() == 0
+                                        && current_visual_line.w + word_width > line_width))
```
Comment (advait-m) [inline src/shape.rs]: Wondering if storing this in a variable and using it across the 4 instances would be helpful 🤔 . Though, downside is that we'd compute it unnecessarily in the case of going into the first if-block maybe.

i.e.
```
word_exceeds_first_line_available_space = (wrap == Wrap::WordOrGlyph
                                        && visual_lines.len() == 0
                                        && current_visual_line.w + word_width > line_width)
```

Might just be simpler to keep it in-line though - I think the comments are very helpful!

**Review by advait-m — APPROVED** [review]
I guess we can't add unit tests for this piece due to the incongruent text overflow bug on the first line?

File: src/shape.rs (hunk 1 above)
Comment (advait-m) [reply src/shape.rs]: (not the best name perhaps above 😅 )

Comment (lucieleblanc) [reply src/shape.rs]: Done, thank you!

**Comment by lucieleblanc** [comment]
Yes, unfortunately. I have the tests running on the Warp side to show this working though.
